### PR TITLE
Changed the recommendation about the LICENSE file for third-party bundles

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | #5599 |

### CODE REVIEW

**Comment by WouterJ** [comment]
:+1:

File: cookbook/bundles/best_practices.rst [hunk 1]
```diff
@@ -102,9 +101,8 @@ that automated tools can rely on:
 * ``README.md``: This file contains the basic description of the bundle and it
   usually shows some basic examples and links to its full documentation (it
   can use any of the markup formats supported by GitHub, such as ``README.rst``);
-* ``Resources/meta/LICENSE``: The full license for the code. The license file
-  can also be stored in the bundle's root directory to follow the generic
-  conventions about packages;
+* ``LICENSE``: The full license for the code. This license file can also be stored
+  in the bundle's ``Resources/meta/`` directory for backwards compatibility reasons;
```
Comment (OskarStark) [inline cookbook/bundles/best_practices.rst]: is it possible to add a symlink for backward compatibility?

then i recommend this to avoid different files

**Comment by OskarStark** [comment]
like your change!

File: cookbook/bundles/best_practices.rst (hunk 1 above)
Comment (WouterJ) [inline cookbook/bundles/best_practices.rst]: What's "backwards compatibility" here? Afaik, no tool relied on this location. And this document is just a recommendation, not a strict standard.

Comment (OskarStark) [inline cookbook/bundles/best_practices.rst]: i think @WouterJ is right...

Comment (javiereguiluz) [inline cookbook/bundles/best_practices.rst]: I agree with Wouter. What if we reword it as follows:

```
* ``LICENSE``: The full contents of the license used by the code. The MIT license is strongly
  encouraged for publicly shared bundles.
```

Comment (WouterJ) [inline cookbook/bundles/best_practices.rst]: :-1: on including a recommendation about MIT. Everyone should be free to choose what they want. If anything, we can add a small sentence to make people aware that a MIT-compatible license is the most usefull for bundle users.

Comment (OskarStark) [inline cookbook/bundles/best_practices.rst]: then your ```most usefull``` is the same like the ```strongly encouraged``` of @javiereguiluz :smile: 

what if we provide a link to known licenses? so people can do their research there, or we say, that symfony and their components itself uses MIT?

Comment (WouterJ) [inline cookbook/bundles/best_practices.rst]: @OskarStark imo, "strongly encouraged" = we almost force you to choose MIT, but you may choose another license if you have very good reasons. "most usefull for your end-users" = you're free to choose another license, but be aware of license conflicts for your users.

And please note that MIT !== MIT-compatible (there are a lot more MIT-compatible licenses).

And yes, I agree that it's a good idea to link to http://choosealicense.com/

Comment (OskarStark) [inline cookbook/bundles/best_practices.rst]: thank you for your detailed infos @WouterJ

Comment (xabbuh) [inline cookbook/bundles/best_practices.rst]: Let's make that change.

Comment (javiereguiluz) [inline cookbook/bundles/best_practices.rst]: I've reworded it as:

```
* ``LICENSE``: The full contents of the license used by the code. Most third-party
  bundles are published under the MIT license, but you can `choose any license`_;
```

**Comment by WouterJ** [comment]
:+1:

**Comment by OskarStark** [comment]
:+1:

**Comment by hhamon** [comment]
:+1:

**Comment by xabbuh** [comment]
Thanks Javier.

**Pull request closed without merging**